collector.go

```diff
@@ -78,14 +62,14 @@ type CollectorEndpointResponse struct {
 	VersionBundles []Bundle `json:"version_bundles"`
 }
 
-func (c *Collector) Collect(ctx context.Context) error {
+func (c *Collector) Collect(ctx context.Context, endpoints []*url.URL) error {
```

Comment:
Why shouldn't `Collect` discover endpoints for itself?

Also if this is used only by cluster-service I think the code should be moved there entirely.

Reply:
I originally though this was shared in many places, but found out later it was only used in cluster-service. I forgot I could move this code there then.